File: Backend/lambdas/POST/createProduct.py
import mysql.connector
import json
  
def lambda_handler(event, context):
    db = mysql.connector.connect(
            host ="127.0.0.1",
            user = "root",
            passwd = "root123",
            database = "dbMain",
            auth_plugin='mysql_native_password'
    )
    data = json.loads(event['body'])
    # preparing a cursor object
    cursorObject = db.cursor()
    
    # creating database
    for register in data['products']:
        try:
            print(register)
            query = "CALL dbMain.uspCreateProduct('"+register['name']+"', '"+register['code']+"', "+str(register['stock'])+", "+str(register['value'])+", '"+register['description']+"', "+str(register['userAddID'])+")"
            print(query)
            cursorObject.execute(query)
            db.commit()
        except:
            db.close()
            response = {
                'status' : 200,
                'body' : json.dumps({'Error': 'Error al insertar registros'})
            }
            return response
    
    response = {
        'status' : 200,
        'body' : json.dumps({'Estado': 'Exito, registros ingresados'})
    }
    db.close()
    return response

File: Backend Fake/lambdas/endpoints/CREATE/createProduct.js
const mysql = require("mysql2");
require("dotenv").config();

module.exports.createProductHandler = async (event) => {
    var con = mysql.createConnection({
        host: process.env.DB_HOST,
        user: process.env.DB_USER,
        password: process.env.DB_PASS,
    });

    con.connect((err) => {
        if (err) throw err;
        console.log("Conectado");
    });

    products = JSON.parse(event.body).products;

    products.forEach((product) => {
        con.query(
            "CALL dbMain.uspCreateproduct('" +
                product.name +
                "', '" +
                product.code +
                "', " +
                product.stock +
                ", " +
                product.value +
                ", '" +
                product.description +
                "'," +
                product.userAddID +
                ")",
            function (err, results, fields) {
                if (err) throw err;
            }
        );
        con.commit(function (err) {
            if (err) {
                con.rollback(function () {
                    throw err;
                });
            }
        });
    });

    return {
        statusCode: 200,
        body: JSON.stringify({
            message: "funcionó",
            input: JSON.parse(event.body),
        }),
    };
};
